openvdb/openvdb/tools/MeshToVolume.h

```diff
@@ -3118,8 +3141,178 @@ traceExteriorBoundaries(FloatTreeT& tree)
 
 ////////////////////////////////////////
 
+template <typename T, Index Log2Dim, typename InteriorTest>
+void
+floodFillLeafNode(tree::LeafNode<T,Log2Dim>& leafNode, const InteriorTest& interiorTest) {
+
+    // Floods fills a single leaf node.
+    // Starts with all voxels in NOT_VISITED.
+    // Final result is voxels in either POSITIVE, NEGATIVE, or NOT_ASSIGNED.
+    // Voxels that were categorized as NEGATIVE are negated.
+    // The NOT_ASSIGNED is all voxels within 0.75 of the zero-crossing.
+    //
+    // NOT_VISITED voxels, if outside the 0.75 band, will query the oracle
+    // to get a POSITIVE Or NEGATIVE sign (with interior being POSITIVE!)
+    //
+    // After setting a NOT_VISITED to either POSITIVE or NEGATIVE, an 8-way
+    // depth-first floodfill is done, stopping at either the 0.75 boundary
+    // or visited voxels.
+    enum VoxelState {
+        NOT_VISITED = 0,
+        POSITIVE = 1,
+        NEGATIVE = 2,
+        NOT_ASSIGNED = 3
+    };
+
+    const auto DIM = leafNode.DIM;
+    const auto SIZE = leafNode.SIZE;
+
+    std::vector<VoxelState> voxelState(SIZE, NOT_VISITED);
+
+    std::vector<std::pair<Index, VoxelState>> offsetStack;
+    offsetStack.reserve(SIZE);
+
+    for (Index offset=0; offset<SIZE; offset++) {
+        const auto value = leafNode.getValue(offset);
+
+        // We do not assign anything for voxel close to the mesh
+        // This condition is aligned with the condition in traceVoxelLine
+        if (abs(value) <= 0.75) {
```

Comment:
std::abs(value). But I'm also curious, is there a significance of the constant 0.75 here?

Reply:
0.75 is a magic constant from traceVoxelLine.
                // Boundary voxel check. (Voxel that intersects the surface)
                if (!(dist > ValueType(0.75))) isOutside = false;

Reply:
This has caught me out in the past too - we should really constexpr inline this constant in `MeshToVolume.h` and use a variable for this